envoy-control-tests/src/main/resources/lua_spec/ingress_handler_spec.lua

```diff
@@ -0,0 +1,227 @@
+    local headers
+    local metadata
+    local ssl
+
+    before_each(function ()
+        headers = {
+            [':status'] = '403'
+        }
+        metadata = {
+            ['envoy.filters.http.rbac'] = {
+                ['shadow_engine_result'] = 'denied'
+            },
+            ['envoy.filters.http.lua'] = {
+                ['request.info.service_name'] = 'service-first',
+                ['request.info.path'] = '/path?query=val',
+                ['request.info.method'] = 'POST',
+                ['request.info.xff_header'] = '127.1.1.3',
+
+            }
+        }
+        ssl = true
+    end)
+
+    describe("should log unauthorized requests:", function ()
+
+        it("https request", function ()
+            -- given
+            local handle = handlerMock(headers, metadata, ssl)
+
+            -- when
+            envoy_on_response(handle)
+
+            -- then
+            assert.spy(handle.logInfo).was_called_with(_,
+              "\nUnauthorized request: method = POST, path = '/path?query=val', "..
+              "clientIp = 127.1.1.3, clientName = service-first, protocol = https, statusCode = 403")
+            assert.spy(handle.logInfo).was_called(1)
+        end)
+
+        it("http request", function ()
+            -- given
+            ssl = false
+            local handle = handlerMock(headers, metadata, ssl)
+
+            -- when
+            envoy_on_response(handle)
+
+            -- then
+            assert.spy(handle.logInfo).was_called_with(_,
+              "\nUnauthorized request: method = POST, path = '/path?query=val', "..
+                "clientIp = 127.1.1.3, clientName = service-first, protocol = http, statusCode = 403")
+            assert.spy(handle.logInfo).was_called(1)
+        end)
+
+        it("allowed & logged request", function ()
+            -- given
+            headers[':status'] = '200'
+            local handle = handlerMock(headers, metadata, ssl)
+
+            -- when
+            envoy_on_response(handle)
+
+            -- then
+            assert.spy(handle.logInfo).was_called_with(_,
+              "\nUnauthorized request: method = POST, path = '/path?query=val', "..
+                "clientIp = 127.1.1.3, clientName = service-first, protocol = https, statusCode = 200")
+            assert.spy(handle.logInfo).was_called(1)
+        end)
+
+        it("request with no metadata fields saved", function ()
+            -- given
+            metadata['envoy.filters.http.lua'] = {}
+            headers = {}
+            local handle = handlerMock(headers, metadata, ssl)
+
+            -- when
+            envoy_on_response(handle)
+
+            -- then
+            assert.spy(handle.logInfo).was_called_with(_,
+              "\nUnauthorized request: method = , path = , clientIp = , clientName = , protocol = https, statusCode = ")
+            assert.spy(handle.logInfo).was_called(1)
+        end)
+
+        it("request with no metadata saved", function ()
```

Comment:
```suggestion
        it("request with no lua filter metadata saved", function ()
```
similarily to one above

Reply:
done